lib/wasi/src/lib.rs

```diff
@@ -337,19 +378,6 @@ lazy_static::lazy_static! {
     static ref CALLER_ID_SEED: Arc<AtomicU32> = Arc::new(AtomicU32::new(1));
 }
 
-/// Returns the current thread ID
-pub fn current_caller_id() -> WasiCallingId {
```

Comment:
This function is currently public.

Is there a usecase for external usage?
If so we can't just remove it.

Reply:
no one is using it externally and it should never have been public - it was a leaky abstraction

Reply:
Yeah then we should make it private. 

We now have decoupled `-wasix` from the other crates for versioning, so we can make breaking changes much more liberally.